crates/rome_js_analyze/src/analyzers/nursery/no_dupe_keys.rs

```diff
@@ -0,0 +1,262 @@
+use crate::utils::batch::JsBatchMutation;
+use rome_analyze::context::RuleContext;
+use rome_analyze::{declare_rule, Ast, Rule, RuleDiagnostic};
+use rome_console::markup;
+use rome_js_syntax::{JsAnyObjectMember, JsObjectExpression};
+use rome_rowan::{AstNode, BatchMutationExt};
+use std::cmp::Ordering;
+use std::collections::HashMap;
+use std::fmt::Display;
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Prevents object literals having more than one property declaration for the same key.
+    /// If an object property with the same key is defined multiple times (except when combining a getter with a setter), only the last definition makes it into the object and previous definitions are ignored, which is likely a mistake.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// const obj = {
+    ///		a: 1,
+    ///		a: 2,
+    /// }
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// const obj = {
+    ///		set a(v) {},
+    ///		a: 2,
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// const obj = {
+    ///		a: 1,
+    ///		b: 2,
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// const obj = {
+    ///		get a() { return 1; },
+    ///		set a(v) {},
+    /// }
+    /// ```
+    ///
+    pub(crate) NoDupeKeys {
+        version: "10.1.0",
```

Comment:
```suggestion
        version: "11.0.0",
```

There are some breaking changes ahead of us :D